effect/jvm/src/main/scala/scalaz/effect/RTS.scala

```diff
@@ -128,6 +128,9 @@ trait RTS {
         future.cancel(true); ()
       }
     }
+
+  final def submitCanceler[E, A](canceler: Throwable => IO[E, A]): Throwable => Unit =
+    th => submit(unsafePerformIO(canceler(th)))
```

Comment:
After thinking about this more, I think we want `unsafePerformIO(canceler(th))`.

The reason is that: for an impure canceler, it's run in the same thread. I don't want that semantic to change merely because the canceler has been made pure. Ideally it should run in the same thread.

Also note that a canceler should always have `Void` as the error type so we know it cannot fail. So you should make that change here because we want a guarantee it won't fail on us (except for non-recoverable error / defect).

Reply:
Got it, I changed it. I also changed the name as is not really submitting anymore. What do you think ? Thanks